project_half_3.py

```diff
@@ -0,0 +1,201 @@
+import os
+import json
+import pandas as pd
+
+def printMenu():
+    print("1. Save a new entry")
+    print("2. Search by ID")
+    print("3. Print ages average")
+    print("4. Print all names")
+    print("5. Print all IDs")
+    print("6. Print all entries")
+    print("7. Print entry by index")
+    print("8. Save all data")
+    print("9. Exit")
+
+def saveNewEntry(dict_by_id, list_of_id):
+    id = input("ID: ")
+
+    id = checkParamIsNumber(id, "ID")

```

Comment:
הרעיון שהפונקציה בודקת אם מספר
1. אתה גם יכול להוסיף positive לשם שלה - כי אם שגיאה היא מחזירה שלילי
2. תמיד יש 2 שורות
`    id = input("ID: ")

    id = checkParamIsNumber(id, "ID")`

במקום זה אפשר לעשות
`id = checkParamIsNumber(input("What is the ID?", "ID")`
ואפילו להתקדם משם - שהאינפוט יקרה בפנים.
זה פשוט קוד שחוזר על עצמו עם ID, גיל וכ'ו